sunpy/map/sources/goes.py

```diff
@@ -0,0 +1,100 @@
+"""
+GOES SXI Map subclass definitions.
+"""
+
+
+from astropy.visualization import AsinhStretch
+from astropy.visualization.mpl_normalize import ImageNormalize
+
+from sunpy.coordinates.sun import B0, angular_radius, earth_distance
+from sunpy.map import GenericMap
+from sunpy.map.sources.source_type import source_stretch
+from sunpy.sun import constants
+
+__all__ = ['SXIMap']
+
+
+class SXIMap(GenericMap):
+    """
+    GOES SXI Image Map
+
+    The GOES Solar X-ray Imager (SXI) provides uninterrupted, full-disk, soft X-ray
+    solar images, with a 1 min cadence and a single-image (adjustable) dynamic
+    range near 100. A set of metallic thin-film filters provides temperature
+    discrimination in the 0.6–6.0 nm bandpass. The spatial resolution of
+    approximately 10 arcsec Full width half maximum (FWHM) is sampled with 5
+    arcsec pixels.
+
+    SXI was first launched on 23 July 2001 on NOAA’s GOES-12 satellite and has
+    obtained uninterrupted images starting on 22 January 2003.
+
+    .. note::
+
+        The second image which is a pixel mask, is ignored and not read into the map.
+
+    References
+    ----------
+    * `Mission Instrument Page <https://sxi.ngdc.noaa.gov/index.html>`__
+    * `Data notes and Fits headers <https://sxi.ngdc.noaa.gov/sxi_data_notes.html>`__
+    * `Official SXI Data Book <https://sxi.ngdc.noaa.gov/images/section06.pdf>`__
+    * `Instrument Reference <https://doi.org/10.1007/s11207-005-7416-x>`__
+    * `SXI Performance <https://doi.org/10.1007/s11207-005-7417-9>`__
+    """
+    # TODO: I added all of them, which might not be very useful.
+    filter_wheel = ["OPEN", "PTHNA", "PTHNB", "PMEDB", "PMEDA", "PTHK",
+                    "BTHNA", "BTHNB", "BMED", "BTHK", "UV", "RDSH"]
+
+    def __init__(self, data, header, **kwargs):
+        # Suggests that the header has been corrected already.
+        if 'DETECTOR' not in header:
```

Comment:
Not sure what the best way to work around this is. I came across this rotating the map, it would create a new map which had a corrected header. Making it error on init.